### ISSUE TITLE
CORS Iframe...

### ISSUE THREAD

**Issue by JasonDD** [koush/vysor.io#597]
Hi 

I'm a premium user and hope to become Enterprise bit have a small issue.... I am aiming to use the iframe sharing option so the view of a device can be seen in a browser but we also want to interrogate the DOM etc... And your current CORS policy won't allow that.

Is there a way to enable us to access the iframe , short of us follow proxying all the requests?

Thanks

Jason

**Comment by koush** [comment]
What are you looking to accomplish exactly?

**Comment by JasonDD** [comment]
We're trying to get the position of where someone clicks on the  view of the screen of the Android device.

Doing this in JavaScript is very possible but not when the CORS policy doesn't allow it across hostnames. As the iframe is served from your domain, the CORS policy needs to be granted by you

**Comment by koush** [comment]
Yeah... and for good reason. Relaxing the CORS policy would allow any site to manipulate the Android device. Let me think about this.

**Comment by koush** [comment]
What do you need the position of the click for? What is your end goal here? What are you doing with that click data?

**Comment by JasonDD** [comment]
I understand the security implications, but I will say these are our devices that we're giving access to access via web browser to authorised people.

Your security aspects of authorising the login, verifying the URL etc obviously recognise that and as such I would say that the CORS policy can be relaxed... If someone has access to the URL they've already been granted permission. 

The end goal we're trying to do it record where the clicks are and that's all about seeing how users interact with the phone, or rather our apps, as part of our UI tests we run.

Bottom line is.... we aren't brilliant Android guys and believed that being able to remotely view our devices by people we authorise and allow to access it (which Vysor does with it's web accessible URL) if iframed on our site.. which you actually say is OK on your Github, we should also be able to query it...

As I said above, from a security stand point, seeing where someone clicks on the phone cant be worse than us (who own the phone(s) ) deciding to give someone access to that phone via a browser

**Comment by koush** [comment]
relaxing CORS would mean any website could put Vysor's website into an iframe and start poking at the user's device. This is not good. The user being authorized is not a concern. The concern is 3rd party websites maliciously accessing a user's Android.

I can see about exposing the raw event feed, but only that.

**Comment by JasonDD** [comment]
I completely understand your point and don't disagree about the cross site issue - hence why there is a CORs policy.

Thank you for your time and if you do expose the event feed, please let us know.. I do have a question though on that topic, as if we have ADB access to the phone, which obviously we do, as it's our phone connected to our computer to enable Vysor, won't we get that via ADB anyway?

or do you mean a different set of data ?

**Comment by koush** [comment]
Yeah, providing a local api would work too. It sounded like you were capturing it from the vysor share client browser (which may be remote).

**Comment by JasonDD** [comment]
Yes we were capturing it from the browser but we can do it locally too.. It seemed sensible, for our use case, to capture it that way.

Thank you for your thoughts.
